src/views/CreateShareStructure.vue

```diff
@@ -1,18 +1,346 @@
+        <v-icon>mdi-plus</v-icon>
+        <span>Add Share Class</span>
+      </v-btn>
+    </div>
+    <v-card flat class="add-share-structure-container" v-if="showShareStructureForm">
+      <ShareStructure
+        v-show="showShareStructureForm"
+        :initialValue="currentShareStructure"
+        :activeIndex="activeIndex"
+        :nextId="nextId"
+        :parentIndex="parentIndex"
+        :shareClasses="shareClasses"
+        @addEditClass="addEditShareClass($event)"
+        @addEditSeries="addEditShareSeries($event)"
+        @removeClass="removeShareClass($event)"
+        @removeSeries="removeSeries"
+        @resetEvent="resetData()"/>
+    </v-card>
+    <ListShareClass v-if="this.shareClasses.length > 0"
+    :shareClasses="shareClasses"
+    :componentDisabled="showShareStructureForm"
+    @editClass="initShareClassForEdit($event)"
+    @removeClass="removeShareClass($event)"
+    @addSeries="initNewShareSeries($event)"
+    @editSeries="editSeries"
+    @removeSeries="removeSeries"/>
+  </section>
 </template>
 
 <script lang="ts">
 // Libraries
 import { Component, Vue } from 'vue-property-decorator'
+import { Action, State } from 'vuex-class'
+
+// Interfaces
+import { ActionBindingIF, FormType, ShareClassIF } from '@/interfaces'
+
+// Components
+import { ListShareClass, ShareStructure } from '@/components/CreateShareStructure'
+
+@Component({
+  components: {
+    ListShareClass,
+    ShareStructure
+  }
+})
+export default class CreateShareStructure extends Vue {
+  // State
+  @State(state => state.stateModel.createShareStructureStep.shareClasses)
+  readonly shareClasses: ShareClassIF[]
+
+  // Actions
+  @Action setShareClasses!: ActionBindingIF
+  @Action setCreateShareStructureStepValidity!: ActionBindingIF
+
+  private newShareClass: ShareClassIF = {
+    id: null,
+    priority: null,
+    type: 'Class',
+    name: '',
+    hasMaximumShares: true,
+    maxNumberOfShares: null,
+    hasParValue: true,
+    parValue: null,
+    currency: 'CAD',
+    hasRightsOrRestrictions: false,
+    series: []
+  }
+
+  private newShareSeries: ShareClassIF = {
+    id: null,
+    priority: null,
+    type: 'Series',
+    name: '',
+    hasMaximumShares: true,
+    maxNumberOfShares: null,
+    hasParValue: true,
+    parValue: null,
+    currency: null,
+    hasRightsOrRestrictions: false
+  }
+
+  private sharesHelpSample: ShareClassIF[] = [{
+    'id': 1,
+    'priority': 0,
+    'type': 'Class',
+    'name': 'Sample Shares',
+    'hasMaximumShares': true,
+    'maxNumberOfShares': 1000,
+    'hasParValue': true,
+    'parValue': 1,
+    'currency': 'CAD',
+    'hasRightsOrRestrictions': true,
+    'series': []
+  }]
+
+  private showShareStructureForm: boolean = false
+  private addEditInProgress: boolean = false
+  private currentShareStructure: ShareClassIF | null = null
+
+  private activeIndex: number = -1
+  private parentIndex: number = -1
+  private nextId: number = -1
+  private helpToggle: boolean = false
```

Comment:
Don't change this but FYI: some linters will complain about the type being redundant here because it's implied by the initial value. So I try to not include type when it's obvious: I mouse over the property and see if it's already typed, otherwise I specify it.